### PR TITLE
Rewindable segment iterators

### PR DESCRIPTION
**Pull request by cmfcmf**
While working at #1394, we noticed that it would be nice to be able to rewind segment iterators. This PR makes all iterators rewindable.

### CODE REVIEW

**Comment by mrzzzrm** [comment]
Which operator(/operation) would realistically be able to profit from this?

**Comment by mrks** [comment]
I also found the need for this when I built the SIMD stuff. In the end, I chose another approach, but for ease of development, I do believe that our iterators should be rather complete, even if we don't use every method. #1515

**Comment by mrks** [comment]
On first look, this appears to be in order.

File: src/lib/storage/run_length_segment/run_length_segment_iterable.hpp [hunk 1]
```diff
@@ -70,11 +70,26 @@ class RunLengthSegmentIterable : public PointAccessibleSegmentIterable<RunLength
       }
     }
 
+    void decrement() {
+      --_current_position;
+
+      if (_current_position <= *(_end_position_it - 1)) {
```
Comment (mrzzzrm) [inline src/lib/storage/run_length_segment/run_length_segment_iterable.hpp]: `_end_position_it - 1` will be an illegal operation if _end_position_it is the `begin` iterator

**Comment by mrzzzrm** [comment]
okay then

File: src/lib/storage/run_length_segment/run_length_segment_iterable.hpp (hunk 1 above)
Comment (cmfcmf) [reply src/lib/storage/run_length_segment/run_length_segment_iterable.hpp]: Good catch! Fixed in https://github.com/hyrise/hyrise/pull/1517/commits/7a2cb883246c0a28f506b909d07e4d95f5ff045b

**Comment by cmfcmf** [comment]
Tests have been added; this should be ready for review now :tada:

**Pull request merged**